packages/KDateTimePicker/KDateTimePicker.vue

```diff
@@ -0,0 +1,502 @@
+              :is-rounded="false"
+              :class="{'selected-option': timeFrame.timeframeText === selectedTimeframe.timeframeText}"
+              :data-testid="'select-timeframe-' + timeFrame.timeframeLength()"
+              appearance="outline"
+              size="medium"
+              @click="changeRelativeTimeframe(timeFrame)"
+            >
+              {{ timeFrame.timeframeText }}
+            </KButton>
+          </div>
+        </div>
+      </div>
+    </template>
+    <template
+      v-if="!hidePopover"
+      #footer>
+      <div class="d-flex justify-content-end">
+        <KButton
+          :is-rounded="false"
+          size="medium"
+          appearance="btn-link"
+          @click="clearSelection()"
+        >
+          Clear
+        </KButton>
+        <KButton
+          :disabled="submitDisabled"
+          :is-rounded="false"
+          size="medium"
+          appearance="btn-link"
+          @click="submitTimeFrame()"
+        >
+          Apply
+        </KButton>
+      </div>
+    </template>
+  </KPop>
+</template>
+
+<script>
+import KButton from '@kongponents/kbutton/KButton.vue'
+import KPop from '@kongponents/kpop/KPop.vue'
+import KSegmentedControl from '@kongponents/ksegmentedcontrol/KSegmentedControl.vue'
+import { format } from 'date-fns'
+
+export default {
+  name: 'KTimePicker',
+  components: {
+    KButton,
+    KPop,
+    KSegmentedControl,
+    DatePicker: () => import('v-calendar/lib/components/date-picker.umd')
+  },
+  props: {
+    defaultMessage: {
+      type: String,
+      required: false,
+      default: 'Select a time range'
+    },
+    calendarMode: {
+      type: String,
+      required: false,
+      default: '',
+      validator: (value) => {
+        return [ '', 'date', 'time', 'dateTime' ].indexOf(value) !== -1
+      }
+    },
+    defaultRelative: {
+      type: Object,
+      required: false,
+      default: () => {}
+    },
+    defaultCustom: {
+      type: Object,
+      required: false,
+      default: () => ({start: '', end: ''})
+    },
+    isRange: {
+      type: Boolean,
+      required: false,
+      default: false
+    },
+    timePeriods: {
+      type: Array,
+      required: false,
+      default: () => [],
+      validator: (valuesArray) => {
+        return valuesArray.every((item) => {
+          return Array.isArray(item.values) && item.values.every((timeframe) => {
+            // Check validity of each timeframe
+            return typeof timeframe.timeframeText === 'string' &&
+              timeframe.timeframeLength !== undefined &&
+              timeframe.start !== undefined &&
+              timeframe.end !== undefined
+          })
+        })
+      }
+    }
+  },
+
+  data () {
+    return {
+      hidePopover: false,
+      tabName: this.getDefaultTabName(),
+      abbreviatedDisplay: this.defaultMessage,
+      fullRangeDisplay: '',
+      selectedCalendarRange: this.defaultCustom,
+      selectedTimeframe: this.timePeriods[0],
```

Comment:
`this.timePeriods` is not a required prop, meaning that the first object in the array may very well not exist. At the least you should provide a fallback value here
```suggestion
      selectedTimeframe: this.timePeriods[0] || '', // or whatever a valid fallback value would be
```